ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/helpers/StructHelper.java

```diff
@@ -0,0 +1,231 @@
+            case "java.util.Date":
+            case "java.util.UUID":
+                return "ds3_str*";
+            case "double":
+                return "double";   // ex: 0.82
+            case "java.lang.Long":
+            case "long":
+                return "uint64_t"; // size_t
+            case "java.lang.Integer":
+            case "int":
+                return "int";
+            case "java.util.Set":
+            case "array":
+                return getResponseTypeName(element.getComponentType()) + "**";
+
+            default:
+                return getResponseTypeName(element.getType()) + "*";
+        }
+    }
+
+    public static boolean isPrimitiveDs3Type(final String type){
+        switch (type) {
+            case "ds3_bool":
+            case "ds3_str*":
+            case "double":
+            case "uint64_t":
+            case "int":
+                return true;
+            default:
+                // any complex sub element such as "com.spectralogic.s3.server.domain.UserApiBean"
+                return false;
+        }
+    }
+
+    public static boolean isPrimitive(final Struct struct) {
+        for (final StructMember member : struct.getVariables()) {
+            if (!isPrimitiveDs3Type(member.getType())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static ImmutableList<StructMember> convertDs3Elements(ImmutableList<Ds3Element> elementsList) throws ParseException {
+        final ImmutableList.Builder<StructMember> builder = ImmutableList.builder();
+        for (final Ds3Element currentElement : elementsList) {
+            builder.add(new StructMember(convertDs3ElementType(currentElement), getNameUnderscores(currentElement.getName())));
+        }
+        return builder.build();
+    }
+
+    public static boolean containsExistingStructs(final Struct struct, final Set<String> existingStructs) {
+        for (final StructMember structMember: struct.getVariables()) {
+            final String testType = StringUtils.stripEnd(structMember.getType(), "*");
+            if (!existingStructs.contains(testType)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static String generateStructMembers(final ImmutableList<StructMember> structMembers) throws ParseException {
+        final StringBuilder outputBuilder = new StringBuilder();
+
+        for (final StructMember member : structMembers) {
+            outputBuilder.append(indent(1)).
+                    append(member.getType()).
+                    append(" ").
+                    append(member.getName()).
+                    append(";").
+                    append("\n");
+        }
+
+        return outputBuilder.toString();
+    }
+
+    public static String generateStructMemberParserLine(final StructMember structMember, final String parserFunction) throws ParseException {
+        final StringBuilder outputBuilder = new StringBuilder();
```

Comment:
Intellij will probably display a message on this saying that you should use plain string concatenation rather than a string builder.